### PR TITLE
Delaware personal credits

### LINKED ISSUE #2441: Delaware personal credits
reference: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2441

### Summary
📝💵🧑‍👩‍👧‍👦

This change adds the personal credits parameters for Delaware's individual income tax. It updates the file `policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml` with the latest values and metadata.

> _`personal_credits`_
> _Delaware updates its tax_
> _falling leaves of change_

### Walkthrough
*  Add personal credits parameters for Delaware individual income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/2486/files?diff=unified&w=0#diff-61e5970509c8b1e601aaae102745b7c6116d254b982c896dec30a27696be382aR1-R20))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+description: Delaware provides these personal credits.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:1]:
```suggestion
description: Delaware provides these personal credits based on filing status.
```

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:1]: need 2021 values

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml (hunk 1 above)
Comment (shuangding415) [reply policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:1]: done

File: policyengine_us/variables/gov/states/de/tax/income/credits/de_personal.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class de_personal(Variable):
```
[the remark this answers is not in the document]
Comment (shuangding415) [reply policyengine_us/variables/gov/states/de/tax/income/credits/de_personal.py:4]: done

File: policyengine_us/variables/gov/states/de/tax/income/credits/de_personal.py
[the remark this answers is not in the document]
Comment (shuangding415) [reply policyengine_us/variables/gov/states/de/tax/income/credits/de_personal.py:1]: done

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    - title: Delaware individual income tax instructions for 2022 line 26a
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+  label: Delaware Personal Credits amount
+  unit: currency-USD
+  period: year
+  breakdown:
+    - filing_status
+JOINT:
+  2021-01-01: 110
+  2022-01-01: 110
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:13]:
```suggestion
```

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+description: Delaware provides these personal credits based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022 line 26a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:4]: add reference for 2021 values

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+  label: Delaware Personal Credits amount
+  unit: currency-USD
+  period: year
+  breakdown:
+    - filing_status
+JOINT:
+  2021-01-01: 110
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:12]:
```suggestion
  1996-01-01: 110
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:12]: for all values

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:15]: need to delete line

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides these personal credits based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022 line 26a
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: The Delaware Code Online - Title 30 - Chapter 11 - Subchapter II - Personal Credits - § 1110. Personal credits
+      href: https://delcode.delaware.gov/title30/c011/sc02/index.html
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:7]:
```suggestion
      href: https://delcode.delaware.gov/title30/c011/sc02/index.html#1110
```

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides these personal credits based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022 line 26a
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: The Delaware Code Online - Title 30 - Chapter 11 - Subchapter II - Personal Credits - § 1110. Personal credits
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:6]:
```suggestion
    - title: The Delaware Code Online - Title 30 - Chapter 11 - Subchapter II - Personal Credits - § 1110(b)(1). Personal credits
```

File: policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+description: Delaware provides these personal credits based on filing status.
+metadata:
+  reference:
+    - title: Delaware individual income tax instructions for 2022 line 26a
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/credits/personal.yaml:5]:
```suggestion
      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=8
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Add https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-01_PaperInteractive.pdf#page=2 as a reference which indicates this is per exemption, i.e. per person. So multiply by `tax_unit_size` rather than indexing on `filing_status`

File: policyengine_us/variables/gov/states/de/tax/income/credits/de_personal_credit.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+class de_personal_credit(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware personal credit"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf"
+    defined_for = StateCode.DE
+
+    def formula(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/credits/de_personal_credit.py:13]: the formula should take the number of exemptions and multiply it by the credit

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**